src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -149,6 +150,11 @@ public ActiveDirectoryDomain(String name, String servers, String site, String bi
     @DataBoundConstructor
     public ActiveDirectoryDomain(String name, String servers, String site, String bindName, String bindPassword, TlsConfiguration tlsConfiguration) {
         this.name = name;
+        // Gives exception if Password is set lees than 14 chars long in FIPS mode.
+        if(FIPS140.useCompliantAlgorithms() && StringUtils.length(bindPassword) < 14) {
+            throw new IllegalArgumentException(Messages.passwordTooShortFIPS());
```

Comment:
maybe better 
```
 throw new hudson.model.Descriptor.FormException(Messages.passwordTooShortFIPS(), "bindPassword");
```

see https://issues.jenkins.io/browse/JENKINS-73771

Reply:
problem with that is that the FormException wasn't available when the first FIPS PRs were opened and then there would be a mismatch between PRs. Previous PR were throwing IllegalArgumentException

Reply:
we did this here https://github.com/jenkinsci/credentials-plugin/pull/558 last week

Reply:
I was thinking of using `FormException` as well in the Maile plugin fix. But it is a checked exception and I didn't want to declare it in the constructor signature. So `IllegalArgumentException` seemed better imo.

Reply:
fair enough, I can live with that :) 
but if not doing it now this will probably never happen though.